Execution of Scala.js fails and hangs in release mode

I'm on Apple M1, java version is:

```
❯ java -version
openjdk version "17" 2021-09-14
OpenJDK Runtime Environment Temurin-17+35 (build 17+35)
OpenJDK 64-Bit Server VM Temurin-17+35 (build 17+35, mixed mode)
```

```
❯ cat js.scala
object Hello extends App {
  var i = 0
  while (i < 5) {
    println(i)
    i += 1
  }
}

❯ scala-cli clean js.scala && scala-cli js.scala --js --js-mode dev
Compiling project (Scala 3.0.2, Scala.JS)
Compiled project (Scala 3.0.2, Scala.JS)
0
1
2
3
4

❯ scala-cli clean js.scala && scala-cli js.scala --js --js-mode release
Compiling project (Scala 3.0.2, Scala.JS)
Compiled project (Scala 3.0.2, Scala.JS)
java.lang.ExceptionInInitializerError
	at com.oracle.svm.core.classinitialization.ClassInitializationInfo.initialize(ClassInitializationInfo.java:315)
	at com.google.javascript.jscomp.DiagnosticGroups.<clinit>(DiagnosticGroups.java:233)
	at com.oracle.svm.core.classinitialization.ClassInitializationInfo.invokeClassInitializer(ClassInitializationInfo.java:375)
	at com.oracle.svm.core.classinitialization.ClassInitializationInfo.initialize(ClassInitializationInfo.java:295)
	at com.google.javascript.jscomp.CompilationLevel.applyFullCompilationOptions(CompilationLevel.java:179)
	at com.google.javascript.jscomp.CompilationLevel.setOptionsForCompilationLevel(CompilationLevel.java:86)
	at org.scalajs.linker.backend.closure.ClosureLinkerBackend.closureOptions(ClosureLinkerBackend.scala:219)
	at org.scalajs.linker.backend.closure.ClosureLinkerBackend.$anonfun$emit$5(ClosureLinkerBackend.scala:96)
	at org.scalajs.logging.Logger.time(Logger.scala:42)
	at org.scalajs.logging.Logger.time$(Logger.scala:40)
	at org.scalajs.logging.ScalaConsoleLogger.time(ScalaConsoleLogger.scala:15)
	at org.scalajs.linker.backend.closure.ClosureLinkerBackend.$anonfun$emit$3(ClosureLinkerBackend.scala:95)
	at scala.Option.map(Option.scala:230)
	at org.scalajs.linker.backend.closure.ClosureLinkerBackend.emit(ClosureLinkerBackend.scala:89)
	at org.scalajs.linker.standard.StandardLinkerImpl.$anonfun$link$2(StandardLinkerImpl.scala:47)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:567)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
Caused by: java.util.MissingResourceException: Can't find bundle for base name com.google.javascript.rhino.Messages, locale en_US
	at java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:2055)
	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1689)
	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1593)
	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1556)
	at java.util.ResourceBundle.getBundle(ResourceBundle.java:932)
	at com.google.javascript.rhino.ErrorFormat.format(ErrorFormat.java:54)
	at com.google.javascript.rhino.SimpleErrorReporter.getMessage(SimpleErrorReporter.java:103)
	at com.google.javascript.rhino.SimpleErrorReporter.getMessage0(SimpleErrorReporter.java:95)
	at com.google.javascript.jscomp.RhinoErrorReporter.<clinit>(RhinoErrorReporter.java:136)
	at com.oracle.svm.core.classinitialization.ClassInitializationInfo.invokeClassInitializer(ClassInitializationInfo.java:375)
	at com.oracle.svm.core.classinitialization.ClassInitializationInfo.initialize(ClassInitializationInfo.java:295)
	... 25 more
```

P.S.: having tons of fun with this tool, absolute incredible addition to Scala tooling!

--- a/modules/cli/src/main/resources/META-INF/native-image/org.virtuslab/scala-cli-core/resource-config.json
+++ b/modules/cli/src/main/resources/META-INF/native-image/org.virtuslab/scala-cli-core/resource-config.json
@@ -18,6 +18,9 @@
       },
       {
         "pattern": "\\Qcommons-logging.properties\\E"
+      },
+      {
+        "pattern": "\\Qcom/google/javascript/jscomp/js/polyfills.txt\\E"
       }
     ]
   },

--- a/modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala
+++ b/modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala
@@ -47,7 +47,7 @@ abstract class RunTestDefinitions(val scalaVersionOpt: Option[String])
     simpleScriptTest()
   }
 
-  def simpleJsTest(): Unit = {
+  def simpleJsTest(extraArgs: String*): Unit = {
     val fileName = "simple.sc"
     val message  = "Hello"
     val inputs = TestInputs(
@@ -61,16 +61,19 @@ abstract class RunTestDefinitions(val scalaVersionOpt: Option[String])
       )
     )
     inputs.fromRoot { root =>
-      val output =
-        os.proc(TestUtil.cli, extraOptions, fileName, "--js").call(cwd = root).out.text().trim
-      expect(output == message)
+      val output = os.proc(TestUtil.cli, extraOptions, fileName, "--js", extraArgs).call(cwd = root).out.text().trim
+      expect(output.linesIterator.toSeq.last == message)
     }
   }
 
-  if (TestUtil.canRunJs)
+  if (TestUtil.canRunJs) {
     test("simple script JS") {
       simpleJsTest()
     }
+    test("simple script JS in release mode") {
+      simpleJsTest("--js-mode", "release")
+    }
+  }
 
   def simpleJsViaConfigFileTest(): Unit = {
     val message = "Hello"

--- a/modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala
+++ b/modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala
@@ -61,7 +61,8 @@ abstract class RunTestDefinitions(val scalaVersionOpt: Option[String])
       )
     )
     inputs.fromRoot { root =>
-      val output = os.proc(TestUtil.cli, extraOptions, fileName, "--js", extraArgs).call(cwd = root).out.text().trim
+      val output = os.proc(TestUtil.cli, extraOptions, fileName, "--js", extraArgs).call(cwd =
+        root).out.text().trim
       expect(output.linesIterator.toSeq.last == message)
     }
   }

--- a/project/settings.sc
+++ b/project/settings.sc
@@ -13,6 +13,8 @@ import mill._, scalalib._
 import scala.collection.JavaConverters._
 import scala.util.Properties
 
+private def isCI = System.getenv("CI") != null
+
 private def withGzipContent[T](gzFile: File)(f: InputStream => T): T = {
   var fis: FileInputStream  = null
   var gzis: GZIPInputStream = null
@@ -211,6 +213,7 @@ trait CliLaunchers extends SbtModule { self =>
       Seq(
         s"-H:IncludeResources=$localRepoResourcePath",
         "-H:-ParseRuntimeOptions",
+        "-H:IncludeResourceBundles=com.google.javascript.jscomp.parsing.ParserConfig",
         s"-H:CLibraryPath=$cLibPath"
       )
     }
@@ -365,7 +368,7 @@ trait CliLaunchers extends SbtModule { self =>
   }
 
   def nativeImage =
-    if (Properties.isLinux && arch == "x86_64")
+    if (Properties.isLinux && arch == "x86_64" && isCI)
       `linux-docker-image`.nativeImage
     else
       `base-image`.nativeImage
@@ -563,8 +566,7 @@ trait LocalRepo extends Module {
     define.Task.sequence(tasks)
   }
 
-  private def vcsState = {
-    val isCI = System.getenv("CI") != null
+  private def vcsState =
     if (isCI)
       T.persistent {
         VcsVersion.vcsState()
@@ -573,7 +575,6 @@ trait LocalRepo extends Module {
       T {
         VcsVersion.vcsState()
       }
-  }
   def localRepoZip = T {
     val repoVer   = vcsState().format()
     val ver       = version()

